sabnzbd/urlgrabber.py

```diff
@@ -325,12 +327,12 @@ def _build_request(url):
     # Detect basic auth
     # Adapted from python-feedparser
     user_passwd = None
-    urltype, rest = urllib.parse.splittype(url)
-    realhost, rest = urllib.parse.splithost(rest)
-    if realhost:
-        user_passwd, realhost = urllib.parse.splituser(realhost)
-        if user_passwd:
-            url = '%s://%s%s' % (urltype, realhost, rest)
+    u = urllib.parse.urlparse(url)
+    if u.username or u.password:
+        user_passwd, host_port = u.netloc.split('@')
```

Comment:
This would fail for `"http://user@gmail.com:password@usenet.nl"`.
Maybe we can even make some unittest for this function.
Probably better to use the `u.username` and `u.password` to build the input for `base64.b64encode`.

Reply:
Damn, how did I miss that...

Anyway, from the http basic auth specs it seems that both user and pass are required, so I'll modify the code to only send basic auth if both are present in the url, and otherwise discard (if, for example, only a user is set but no pass). The old code did pass such fringe cases along though.

The thought of unittests did pass my mind when running some test urls, but without a clear idea of what to accept and what not I didn't yet write any.